cluster.go

```diff
@@ -175,6 +180,8 @@ func NewCluster(
 	}
 	c.setupRPCClients()
 
+	c.unauthPeerTimeCache = timecache.NewTimeCache(c.config.UnauthorizedRequestCacheSpan)
```

Comment:
move this up. Can initialize directly in the cluster object. By doing it here you open a race condition where the API can send requests but the cache is still nil.